lib/autoupdate/agent/process.go

```diff
@@ -246,6 +294,56 @@ func tickFile(ctx context.Context, path string, ch chan<- int, tickC <-chan time
 	}
 }
 
+// waitForReady polls the SocketPath unix domain socket with HTTP requests.
+// If one request returns 200 before the timeout, the service is considered ready.
+func (s SystemdService) waitForReady(ctx context.Context, pid int, tickC <-chan time.Time) error {
+	for {
+		ready, err := s.Ready.GetReadiness(ctx)
+		if err == nil &&
+			ready.Ready &&
+			equalOrZero(ready.PID, pid) {
+			return nil
+		}
+		select {
+		case <-ctx.Done():
+			var netError net.Error
+			if trace.IsNotFound(err) {
+				s.Log.WarnContext(ctx, "Socket appears to be missing readiness endpoint.", unitKey, s.ServiceName)
+				return nil
+			}
+			if errors.Is(err, os.ErrNotExist) ||
+				errors.Is(err, syscall.EINVAL) ||
+				errors.Is(err, os.ErrInvalid) ||
+				errors.As(err, &netError) {
```

Comment:
If we're not going to use the error value and we just need to check its type (also note how we're giving `errors.As` a pointer to a `*net.Error`, which is the actual type implementing `error`):

```suggestion
			if trace.IsNotFound(err) {
				s.Log.WarnContext(ctx, "Socket appears to be missing readiness endpoint.", unitKey, s.ServiceName)
				return nil
			}
			if errors.Is(err, os.ErrNotExist) ||
				errors.Is(err, syscall.EINVAL) ||
				errors.Is(err, os.ErrInvalid) ||
				errors.As(err, new(*net.Error)) {
```

Reply:
`net.Error` is an interface. `new(net.Error)` (`*net.Error`) works here,  but `new(*net.Error)` (`**net.Error`) panics due to the nil.